lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1312,18 +1301,27 @@ def cluster_name_from_kube_env(kube_env)
       instance_prefix
     end
 
+    def time_from_seconds_and_nanos(ts_secs, ts_nanos)
+      Time.at((Integer ts_secs) + (Integer ts_nanos) / 1_000_000_000.0)
```

Comment:
This converts nanoseconds to a double, which [loses precision](https://ruby-doc.org/core-2.2.0/Time.html#method-i-nsec). You could instead use the 2-argument version:
```ruby
     Time.at((Integer ts_secs), (Integer ts_nanos) / 1000.0)
```